### PR TITLE
gateware/ice40: add mux stage and direct register drive to LED block

### PR DESCRIPTION
**Pull request by TomKeddie**
As requested in https://github.com/im-tomu/foboot/issues/20

### CODE REVIEW

**Comment by TomKeddie** [comment]
mithro: please help me with style (especially CAPS where not appropriate) and the issue mentioned on irc. any suggestions on how to get migen to emit binary?  Am seeing "ERROR: expected configuration string starting with '0b' for parameter 'RGB0_CURRENT' on cell 'SB_RGBA_DRV'" because my binary was evaluate to a decimal constant (.RGB0_CURRENT(2'd3).  Thanks.

**Comment by TomKeddie** [comment]
Note that this is not register compatible with foboot impl, I opted for a separate mux register to make the bits align.  I'm planning an emulation of this block for other platforms and want it to be for _n_ leds so the mismatching bits don't work too well.

**Comment by mithro** [comment]
I think you need an str in there somewhere. Poke me tomorrow when I'm not in a bar. :-)

**Comment by mithro** [comment]
@TomKeddie It seems like the str fixed the issue?

**Comment by TomKeddie** [comment]
mithro: yes, thanks.  I feel like I need to test this code more but am looking for a nod that this is what you intended and is moving in the right direction.  Thanks.

**Comment by TomKeddie** [comment]
mithro: when you have a moment please give this a once over, thanks.

File: gateware/ice40.py [hunk 1]
```diff
@@ -115,29 +115,41 @@ def __init__(self, width=32, size=64*1024):
...
             p_CURRENT_MODE = "0b1",
-            p_RGB0_CURRENT = "0b000011",
-            p_RGB1_CURRENT = "0b000001",
-            p_RGB2_CURRENT = "0b000011",
+            p_RGB0_CURRENT = led_defs[rev_map[0]]["CURRENT"],
+            p_RGB1_CURRENT = led_defs[rev_map[1]]["CURRENT"],
+            p_RGB2_CURRENT = led_defs[rev_map[2]]["CURRENT"],
         )
 
+        self.comb += [If(self.mux.storage[0], led_drv[led_defs["R"]["SIGNAL_INDEX"]].eq(self.raw.storage[0])).Else(led_drv[led_defs["R"]["SIGNAL_INDEX"]].eq(pwm_src[0])),
```
Comment (mithro) [inline gateware/ice40.py]: Newline here.

**Comment by mithro** [comment]
Can you rebase this on master?

**Comment by TomKeddie** [comment]
replaced by https://github.com/timvideos/litex-buildenv/pull/180

**Pull request closed without merging**